src/H5FDsubfiling/H5FDsubfiling.c

```diff
@@ -564,6 +564,114 @@ H5Pget_fapl_subfiling(hid_t fapl_id, H5FD_subfiling_config_t *config_out)
     FUNC_LEAVE_API(ret_value)
 } /* end H5Pget_fapl_subfiling() */
 
+herr_t
+H5FDsubfiling_get_file_mapping(hid_t file_id, char ***filenames, size_t *len)
+{
+    subfiling_context_t *sf_context    = NULL;
+    H5FD_t              *driver        = NULL;
+    H5F_t               *file_ptr      = NULL;
+    char               **filenames_arr = NULL;
+    char                *filepath      = NULL;
+    char                *subfile_dir   = NULL;
+    char                *base          = NULL;
+    int                  num_subfiles  = 0;
+    int                  num_digits    = 0;
+    herr_t               ret_value     = SUCCEED;
+
+    FUNC_ENTER_API(FAIL)
+
+    if (file_id < 0)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid file ID");
+    if (!filenames)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "`filenames` was NULL");
+    if (!len)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "`len` was NULL");
+
+    *filenames = NULL;
+    *len       = 0;
+
+    if (NULL == (file_ptr = H5VL_object(file_id)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid file ID");
```

Comment:
We should probably add a check right after this line to make sure that the driver being used is actually Subfiling and return an error if not. Something like

```C
if (H5FD_SUBFILING != H5F_get_driver_id(file_ptr))
    HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "file is not using Subfiling VFD");
```